calico_containers/pycalico/ipam.py

```diff
@@ -112,12 +117,17 @@ def unassign_address(self, pool, address):
         Unassign an IP from a pool.
 
         :param IPPool or IPNetwork pool: The pool that the assignment is from.
+        If the pool is None, get the pool from datastore
         :param IPAddress address: The address to unassign.
-
         :return: True if the address was unassigned, false otherwise. An
         exception is thrown for any error conditions.
         :rtype: bool
         """
+        pool = pool or self.get_pool(address)
```

Comment:
I don't think calico knows how to handle overlapping pools anywhere. Might be worth raising a backlog issue to define which pool should be chosen (probably the most specific) and implementing it.